shared/rangerclient/client.go

```diff
@@ -0,0 +1,53 @@
+package rangerclient
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/viper"
+	cnquery_config "go.mondoo.com/cnquery/apps/cnquery/cmd/config"
+	"go.mondoo.com/ranger-rpc"
+)
+
+type RangerClientOpts struct {
+	Proxy string
+}
+
+// NewRangerClient will set up the underlyig ranger client
+// with the appropriate proxy if needed.
+func NewRangerClient() (*http.Client, error) {
+	proxy := getMondooAPIProxy()
+
+	rangerClient, err := ranger.HttpClient(&ranger.HttpClientOpts{
+		Proxy: proxy,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return rangerClient, nil
+}
+
+// getMondooAPIProxy will in order of precedence use the proxy info found in
+// 1) MONDO_API_PROXY env var
+// 2) the --api-proxy CLI parameter
+// 3) the api_proxy setting in the config file
+func getMondooAPIProxy() string {
```

Comment:
It feels strange to have this here, this belongs into the apps/cnquery/cmd package

Reply:
It should be a helper in the opts instead similar to the upstream helper